modules/virtualization/microvm/appvm.nix

```diff
@@ -0,0 +1,176 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  configHost = config;
+  cfg = config.ghaf.virtualization.microvm.appvm;
+  waypipe-ssh = pkgs.callPackage ../../../user-apps/waypipe-ssh {};
+
+  makeVm = {vm}: let
+    hostname = "vm-" + vm.name;
+    appvmConfiguration = {
+      imports = [
+        ({
+          lib,
+          config,
+          ...
+        }: {
+          ghaf = {
+            users.accounts.enable = lib.mkDefault configHost.ghaf.users.accounts.enable;
+            profiles.graphics.enable = true;
+
+            development = {
+              # NOTE: SSH port also becomes accessible on the network interface
+              #       that has been passed through to NetVM
+              ssh.daemon.enable = lib.mkDefault configHost.ghaf.development.ssh.daemon.enable;
+              debug.tools.enable = lib.mkDefault configHost.ghaf.development.debug.tools.enable;
+            };
+          };
+
+          users.users.${configHost.ghaf.users.accounts.user}.openssh.authorizedKeys.keyFiles = ["${waypipe-ssh}/keys/waypipe-ssh.pub"];
+
+          networking.hostName = hostname;
+          system.stateVersion = lib.trivial.release;
+
+          nixpkgs.buildPlatform.system = configHost.nixpkgs.buildPlatform.system;
+          nixpkgs.hostPlatform.system = configHost.nixpkgs.hostPlatform.system;
+
+          networking = {
+            enableIPv6 = false;
+            interfaces.ethint0.useDHCP = false;
+            firewall.allowedTCPPorts = [22];
+            firewall.allowedUDPPorts = [67];
+            useNetworkd = true;
+          };
+
+          environment.systemPackages = [
+            pkgs.waypipe
+          ];
+
+          microvm = {
+            mem = vm.ramMb;
+            vcpu = vm.cores;
+            hypervisor = "qemu";
+            qemu.bios.enable = true;
+            storeDiskType = "squashfs";
+            interfaces = [
+              {
+                type = "tap";
+                id = hostname;
+                mac = vm.macAddress;
+              }
+            ];
+          };
+
+          networking.nat = {
+            enable = true;
+            internalInterfaces = ["ethint0"];
+          };
+
+          # Set internal network's interface name to ethint0
+          systemd.network.links."10-ethint0" = {
+            matchConfig.PermanentMACAddress = vm.macAddress;
+            linkConfig.Name = "ethint0";
+          };
+
+          systemd.network = {
+            enable = true;
+            networks."10-ethint0" = {
+              matchConfig.MACAddress = vm.macAddress;
+              addresses = [
+                {
+                  # IP-address for debugging subnet
+                  addressConfig.Address = vm.ipAddress;
+                }
+              ];
+              routes = [
+                {routeConfig.Gateway = "192.168.101.1";}
```

Comment:
What if we'd declare the `netvm` gateway address, and the whole internal network, following this idea https://github.com/unbel13ver/ghaf-lib/blob/main/my-hardware/lenovo-t14.nix 
It would make declaring the subnetwork and VM-specific IP addresses clearer and centralized instead of distributed across the ghaf VMs declarations.